knowledge-base/oracle-error-ora-01008-on-report-preview .md

```diff
@@ -0,0 +1,43 @@
+---
+title: Oracle error ORA-01008 - Not all variables bound 
+description: Oracle error ORA-01008 - Not all variables bound 
+type: troubleshooting
+page_title: Oracle error ORA-01008 - Not all variables bound 
+slug: oracle-error-ora-01008-on-report-preview 
+ticketid: 1526844
+res_type: kb
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product</td>
+			<td>Progress® Telerik® Reporting</td>
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+
+On previewing the report in a report viewer, an Oracle error **'ORA-01008: Not all variables are bound'** is thrown. Sometimes, the report will display without errors in the [Standalone Report Designer](../standalone-report-designer) but the error gets thrown when displaying that same report in a report viewer. 
+
+Usually, the error occurs when there is **one or more** SQL parameter/s used at **two or more places** in the query used by the [SqlDataSource component](../sqldatasource) and the data provider being used is the **Oracle.ManagedDataAccess** provider.
+
```

Comment:
I believe it will be useful to mention two things here: 
1) that not **any** query can cause that exception, but only a specific queries containing nested select clauses.
2) the problem is reproducible in apps that do not use Telerik Reporting at all, so the clients can verify if they have this problem by simply extracting the query and execute it separately.